projects/components/src/action-menu/action-search.provider.spec.ts

```diff
@@ -0,0 +1,99 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { MockTranslationService } from '@vcd/i18n';
+import { ActionItem } from '../common/interfaces/index';
+import { SpotlightSearchService } from '../spotlight-search/index';
+import { ActionSearchProvider } from './action-search.provider';
+
+interface HasActionSearchProvider {
+    actionSearchProvider: ActionSearchProvider<any, any>;
+}
+
+const MOCK_DATA: ActionItem<any, any>[] = [
+    {
+        textKey: 'Contextual 1',
+        handler: () => console.log('Contextual 1'),
+    },
+    {
+        textKey: 'power.actions',
+        children: [
+            {
+                textKey: 'Start',
+                handler: () => console.log('Starting '),
+            },
+            {
+                textKey: 'Stop',
+                handler: () => console.log('Stopping '),
+            },
+        ],
+    },
+];
+
+describe('ActionSearchProvider', () => {
+    beforeEach(function(this: HasActionSearchProvider): void {
+        this.actionSearchProvider = new ActionSearchProvider(
+            new SpotlightSearchService(),
+            new MockTranslationService()
+        );
+    });
+    describe('actions', () => {
+        it('updates the flattened list of available actions when set', function(this: HasActionSearchProvider): void {
```

Comment:
I don't think that the test should explain the inner structure. They should explain what is the behavior of the item in terms of its public behavior